backend/src/services/identity/identity-service.ts

```diff
@@ -115,7 +115,7 @@ export const identityServiceFactory = ({
           { identityId: id },
           {
             role: customRole ? OrgMembershipRole.Custom : role,
-            roleId: customRole?.id
+            roleId: customRole?.id || null
```

Comment:
If customRole is not defined then it will be `undefined`, right? I would think Knex would treat `undefined` like null under the hood

Reply:
The difference is undefined in update means there is nothing and knexjs ignore that field.

Where as null is something else in SQL. Null is empty in SQL world. So in update if there customRole is empty then we set that field as null. 

The issue was before consider user set a custom role and then switched back to a predefined the customRole would be undefined. So the `roleId` was undefined, which basically made knexjs just ignore it rather than unsetting it causing tangling foreign key